src/celeritas/decay/interactor/MuDecayInteractor.hh

```diff
@@ -0,0 +1,235 @@
+                   value_as<Energy>(particle.total_energy())}
+    , max_energy_(real_type{0.5} * shared_.muon_mass.value()
+                  - shared_.electron_mass.value())
+{
+    CELER_EXPECT(shared_);
+    CELER_EXPECT(particle.particle_id() == shared_.mu_minus_id
+                 || particle.particle_id() == shared_.mu_plus_id);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the muon decay.
+ */
+template<class Engine>
+CELER_FUNCTION Interaction MuDecayInteractor::operator()(Engine& rng)
+{
+    // Allocate secondaries
+    Secondary* secondaries = allocate_(1);
+    if (secondaries == nullptr)
+    {
+        // Failed to allocate secondaries
+        return Interaction::from_failure();
+    }
+
+    real_type electron_energy_frac{};
+    real_type electron_nu_energy_frac{};
+    do
+    {
+        do
+        {
+            electron_nu_energy_frac = generate_canonical(rng);
+        } while (RejectionSampler(
+            electron_nu_energy_frac * (real_type{1} - electron_nu_energy_frac),
+            real_type{0.25})(rng));
+
+        electron_energy_frac = generate_canonical(rng);
+    } while (electron_nu_energy_frac + electron_energy_frac < real_type{1});
+
+    // Momentum of secondaries at rest frame
+    auto charged_lep_energy
+        = this->calc_momentum(electron_energy_frac, shared_.electron_mass);
+
+    // Decay isotropically in rest frame and boost secondaries to the lab frame
+    auto charged_lep_4vec = this->to_lab_frame(IsotropicDistribution{}(rng),
+                                               charged_lep_energy,
+                                               shared_.electron_mass);
+
+    // Return charged lepton only
+    Interaction result = Interaction::from_absorption();
+    result.secondaries = {secondaries, 1};
+    result.secondaries[0].particle_id = sec_id_;
+    // Interaction stores kinetic energy; FourVector stores total energy
+    result.secondaries[0].energy
+        = Energy{charged_lep_4vec.energy - shared_.electron_mass.value()};
+    result.secondaries[0].direction = make_unit_vector(charged_lep_4vec.mom);
+
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Boost secondary to the lab frame.
+ *
+ * \note This assumes the primary to be at rest and, thus, there is no need
+ * to perform an inverse boost of the primary at the CM frame.
+ */
+CELER_FUNCTION FourVector MuDecayInteractor::to_lab_frame(Real3 const& dir,
+                                                          MevMomentum momentum,
+                                                          Mass mass) const
+{
+    CELER_EXPECT(is_soft_unit_vector(dir));
+    CELER_EXPECT(momentum > zero_quantity());
+    CELER_EXPECT(mass >= zero_quantity());
+
+    Real3 p = dir * momentum.value();
+    FourVector lepton_4vec{
+        p, std::sqrt(ipow<2>(norm(p)) + ipow<2>(mass.value()))};
```

Comment:
```suggestion
        p, std::sqrt(dot_product(p, p) + ipow<2>(mass.value()))};
```

Reply:
I think:
```suggestion
        p, norm(p) + ipow<2>(mass.value()))};
```

Reply:
But this leads to $$E = p + m^2$$...?

Reply:
Sorry I misread the parens there! Silly me. But instead of `dot_product` and such, use `momentum.value()`!